mne/viz/tests/test_epochs.py

```diff
@@ -100,6 +100,40 @@ def test_plot_epochs():
     plt.close('all')
 
 
+def test_plot_epochs():
+    """Test epoch plotting"""
+    import matplotlib.pyplot as plt
+    epochs = _get_epochs()
+    epochs.plot(scalings=None, fig_title='Epochs', trellis=False)
+    plt.close('all')
+    epochs[0].plot(picks=[0, 2, 3], scalings=None, trellis=False)
+    plt.close('all')
+    fig = epochs.plot(trellis=False)
+    fig.canvas.key_press_event('left')
+    fig.canvas.key_press_event('right')
+    fig.canvas.key_press_event('up')
+    fig.canvas.key_press_event('down')
+    fig.canvas.key_press_event('pageup')
+    fig.canvas.key_press_event('pagedown')
+    fig.canvas.resize_event()
+    fig.canvas.close_event()  # closing and epoch dropping
+    plt.close('all')
+    with warnings.catch_warnings(record=True):
+        fig = epochs.plot(trellis=False)
+        # test mouse clicks
+        x = fig.get_axes()[0].get_xlim()[1] / 2
+        y = fig.get_axes()[0].get_ylim()[0] / 2
+        data_ax = fig.get_axes()[0]
+        _fake_click(fig, data_ax, [x, y], xform='data')  # mark a bad epoch
```

Comment:
you should add a test that checks if the bad epochs are actually dropped after the window is closed